cpp-lite/include/mxnet-lite/ndarray.h

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef MXNET_LITE_NDARRAY_H_
+#define MXNET_LITE_NDARRAY_H_
+
+#include <map>
+#include <string>
+#include <vector>
+
+#include <dmlc/logging.h>
+#include <mxnet/c_api.h>
+
+#include <mxnet-lite/context.h>
+
+namespace mxnet {
+namespace lite {
```

Comment:
Having a different namespace for lite - would that wean that they wouldn't be compatible with the "full" NDArray? I see advantages in keeping the same but reduced interface for the full and lite version and make 1 inherit from another.